src/connection.rs

```diff
@@ -156,12 +157,12 @@ fn tcp_stream(url: &Url, opts: &TcpOptions) -> Result<TcpStream, MemcacheError>
 }
 
 impl Connection {
-    pub(crate) fn connect(url: &Url) -> Result<Self, MemcacheError> {
-        let transport = Transport::from_url(url)?;
+    pub(crate) fn connect(url: Url) -> Result<Self, MemcacheError> {
```

Comment:
Why the url is passing by value instead of by reference now?

Reply:
My bad, I wanted to remove the clone [here](https://github.com/aisk/rust-memcache/pull/89/files#diff-c80c453b374ae714374e0f9d9d86ba1cL164). But, wasn't thinking properly. Fixed now.